modules/repository/src/main/java/org/mule/runtime/module/repository/interal/DefaultRepositoryService.java

```diff
@@ -0,0 +1,134 @@
+
+public class DefaultRepositoryService implements RepositoryService
+{
+
+    private static final String REPOSITORY_FOLDER = "repository";
+    private static final String DEFAULT_REPOSITORY_TYPE = "default";
+
+    private static final Logger logger = LoggerFactory.getLogger(DefaultRepositoryService.class);
+
+    private final RepositorySystem repositorySystem;
+    private final DefaultRepositorySystemSession repositorySystemSession;
+    private List<RemoteRepository> remoteRepositories;
+    private File dependenciesFolder;
+
+    public DefaultRepositoryService()
+    {
+        repositorySystem = new SpiRepositorySystemFactory().createRepositorySystem();
+        createRepositoryFolderIfDoesNotExists();
+        collectRemoteRepositories();
+        repositorySystemSession = new DefaultRepositorySystemSession();
+        repositorySystemSession.setLocalRepositoryManager(repositorySystem.newLocalRepositoryManager(repositorySystemSession, new LocalRepository(dependenciesFolder)));
+    }
+
+    private void collectRemoteRepositories()
+    {
+        String remoteRepositories = System.getProperty(MULE_REMOTE_REPOSITORIES_PROPERTY, "");
+        String[] remoteRepositoriesArray = remoteRepositories.split(",");
+        this.remoteRepositories = new ArrayList<>();
+        for (String remoteRepository : remoteRepositoriesArray)
+        {
+            if (!remoteRepositories.trim().equals(""))
+            {
+                this.remoteRepositories.add(new RemoteRepository.Builder(remoteRepository, DEFAULT_REPOSITORY_TYPE, remoteRepositories.trim()).build());
+            }
+        }
+    }
+
+    private void createRepositoryFolderIfDoesNotExists()
+    {
+        resolveDependenciesFolder();
+        createDependenciesFolder();
+    }
+
+    private void createDependenciesFolder()
+    {
+        if (!dependenciesFolder.exists() && !dependenciesFolder.mkdirs())
+        {
+            throw new RuntimeException("Could not create dependencies folder with path " + dependenciesFolder.getAbsolutePath());
+        }
+    }
+
+    private void resolveDependenciesFolder()
+    {
+        String userDefinedDependenciesFolder = System.getProperty(MULE_REPOSITORY_FOLDER_PROPERTY);
+        if (userDefinedDependenciesFolder != null)
+        {
+            dependenciesFolder = new File(userDefinedDependenciesFolder);
+        }
+        else
+        {
+            dependenciesFolder = new File(MuleContainerBootstrapUtils.getMuleLibDir(), REPOSITORY_FOLDER);
+        }
+        if (logger.isDebugEnabled())
+        {
+            logger.debug("Using dependencies folder " + dependenciesFolder.getAbsolutePath());
+        }
+    }
+
+    @Override
+    public File lookupBundle(BundleDescriptor bundleDescriptor)
+    {
+        try
+        {
+            if (remoteRepositories.isEmpty())
+            {
+                throw new RepositoryServiceDisabledException("Repository service has not been configured so it's disabled. " +
```

Comment:
because returning null is misleading. This should failed if gets used in the wrong environment and not fail silently.